### PR TITLE
feat: add torus auth method

### PR DESCRIPTION
**Pull request by vasa-develop**
Added Torus as a default auth method for mobile and desktop browsers.
This enables a user to log in with their Google, Facebook, Twitch, Reddit or Discord account via `bnc-onboard` widget.

Here is an awesome preview:
bnc-onboard-torusonline-video-cu

License: MIT
Signed-off-by: Vaibhav Saini <vasa@dappkit.io>

### CODE REVIEW

**Comment by vasa-develop** [comment]
You may need to tweak the `get()` method of the `balance` part. 
Right now, the decimal point is not in the right place.

**Comment by YZhenY** [comment]
Hi there @aaronbarnardsound. Would love to get this merged in, got several dapps keen on using blocknative with us in the pipeline. Let me know what we can do to help!

@vasa-develop Thanks for the PR, appreciate it

**Comment by cmeisl** [comment]
Thank you @vasa-develop for the PR. @YZhenY we are taking a look and expect to merge early next week. (We're a little slower last and this week due to holiday schedules).

**Comment by aaronbarnardsound** [comment]
@vasa-develop @YZhenY I will be checking this out today, will keep you updated.

**Comment by aaronbarnardsound** [comment]
@vasa-develop The Torus implementation is looking pretty good. I took the opportunity to refactor some of the interfaces to allow more flexibility when adding wallets as it was needed in this PR for it all to work nicely. I sorted out the `balance` issue and added some validation.

@YZhenY There is one problem I am having with Torus, when the `logout` function is called on the Torus instance, it throws an error: `User has not logged in yet` even though this is called after logging in successfully. Could you look in to that issue on your end as that will need to be resolved before we can merge this PR.

**Comment by aaronbarnardsound** [comment]
@YZhenY Have you had a chance to have a look at the logout issued mentioned above?

**Comment by YZhenY** [comment]
@aaronbarnardsound been swamped with our recent release. will find time take a look at this today

**Comment by YZhenY** [comment]
@iamonuwa could you look into this

**Comment by iamonuwa** [comment]
Right away. Thanks

File: src/modules/select/wallets/torus.ts [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+      return {
+        provider,
+        instance,
+        interface: {
+          name: 'Torus',
+          connect: async () => {
+            let result = await instance.login({ verifier: loginMethod })
+            return { message: result[0] }
+          },
+          disconnect: () => instance.logout(),
```
[the remark this answers is not in the document]
Comment (aaronbarnardsound) [reply src/modules/select/wallets/torus.ts]: Sweet will try that now. Thanks!

**Pull request merged**